src/urllib3/response.py

```diff
@@ -986,6 +986,9 @@ def read(
 
         if amt is None:
             data = self._decode(data, decode_content, flush_decoder)
+            if len(self._decoded_buffer) != 0:
+                data = self._decoded_buffer.get_all() + data
```

Comment:
The buffer should stay untouched if `decode_content` is `False`, so the check from lines 996–1001 should be used regardless of the `amt` value